robot/framesystem/framesystem.go

```diff
@@ -242,6 +208,68 @@ func (svc *frameSystemService) TransformPose(
 	return pose, nil
 }
 
+// AllCurrentInputs will get present inputs for a framesystem from a robot and return a map of those inputs, as well as a map of the
+// InputEnabled resources that those inputs came from.
+func (svc *frameSystemService) AllCurrentInputs(
+	ctx context.Context,
+) (map[string][]referenceframe.Input, map[string]referenceframe.InputEnabled, error) {
+	fs, err := svc.FrameSystem(ctx, []*referenceframe.LinkInFrame{})
+	if err != nil {
+		return nil, nil, err
+	}
+	input := referenceframe.StartPositions(fs)
+
+	// build maps of relevant components and inputs from initial inputs
+	allOriginals := map[string][]referenceframe.Input{}
+	resources := map[string]referenceframe.InputEnabled{}
+	for name, original := range input {
+		// skip frames with no input
+		if len(original) == 0 {
+			continue
+		}
+
+		// add component to map
+		allOriginals[name] = original
+		components := robot.AllResourcesByName(svc.r, name)
+		if len(components) != 1 {
+			return nil, nil, fmt.Errorf("got %d resources instead of 1 for (%s)", len(components), name)
```

Comment:
[nit] This error shows up in a couple places, can we make it a named function in this package?